### DISCUSSION TITLE
Feature: Gallery Page

### DISCUSSION THREAD

**Discussion by GrantBirki** [arthelokyo/astrowind#275] [Ideas]
# Gallery Page

Photographers, small businesses, and all sorts of folks have great photos and they may want to show them off in a `/gallery` page. I went about building this out myself in a hacky `.jsx` manner and think a properly built one would be a great idea!

## Example 📸 

![Image](https://github.com/onwidget/astrowind/assets/23362539/e448f76d-3970-48c4-906b-177dbe4529f5)

The source code for this gallery can be found:

- [here](https://github.com/GrantBirki/astrowind-hard-fork/blob/be29d05cc0f3fe04e37ade9d38c653ed55c6cf53/src/components/widgets/Gallery.jsx)
- [and here](https://github.com/GrantBirki/astrowind-hard-fork/blob/be29d05cc0f3fe04e37ade9d38c653ed55c6cf53/src/pages/gallery.astro)

## Live Demo

A live demo of my site where this lives can be found [here](https://astro-demo.birki.io/gallery/)

---

related: https://github.com/onwidget/astrowind/issues/271#issue-1908178585

**Comment by prototypa** [comment]
Hello @GrantBirki.

Thank you very much for the contribution. I have moved this issue to **discussion/ideas** since you are giving a suggestion.

Until now I have not wanted to put _React_ in the base code so that everyone can enrich **AstroWind** to their liking. There is a large group of people that uses _React_, another large group that uses _Vue_, many other _Alpinejs_, etc. etc.

I'll think about it. Maybe we can do something like a small _Marketplace_ with components for AstroWind. Or install it through the command line. I have needed this for a long time for the projects I do with AstroWind and components like this gallery are usually very useful and give a lot of value to the template.

Let's keep in touch and don't hesitate to write anything that comes to your mind.

**Reply by GrantBirki** [reply]
> Until now I have not wanted to put React in the base code so that everyone can enrich AstroWind to their liking

I think you should keep it that way for as long as you can. As soon as I integrated React into my fork of your project, everything started to get more complicated. Dependency issues, weird console errors, UI bugs, etc. Again, I am no front end developer but React certainly made my life more difficult. As was able to add cool new pages but at the cost of dependency troubles and UI bugs.

That being said, I think it would be awesome to have these `/gallery` pages created but perhaps in pure **Astro**. Feel free to use my code / design as inspiration should you chose to go down this path and add a cool new feature!

**TL;DR**: I make the `/gallery` page on my own fork of Astrowind in React. I did not enjoy the experience. If you want to re-create what I did and include it in this template, you should use native Astro. Good luck!
